hrms/hr/doctype/leave_encashment/leave_encashment.py

```diff
@@ -40,28 +46,31 @@ def on_submit(self):
 		if not self.leave_allocation:
 			self.db_set("leave_allocation", self.get_leave_allocation().get("name"))
 
-		additional_salary = frappe.new_doc("Additional Salary")
-		additional_salary.company = frappe.get_value("Employee", self.employee, "company")
-		additional_salary.employee = self.employee
-		additional_salary.currency = self.currency
-		earning_component = frappe.get_value("Leave Type", self.leave_type, "earning_component")
-		if not earning_component:
-			frappe.throw(_("Please set Earning Component for Leave type: {0}.").format(self.leave_type))
-		additional_salary.salary_component = earning_component
-		additional_salary.payroll_date = self.encashment_date
-		additional_salary.amount = self.encashment_amount
-		additional_salary.ref_doctype = self.doctype
-		additional_salary.ref_docname = self.name
-		additional_salary.submit()
-
-		# Set encashed leaves in Allocation
-		frappe.db.set_value(
-			"Leave Allocation",
-			self.leave_allocation,
-			"total_leaves_encashed",
-			frappe.db.get_value("Leave Allocation", self.leave_allocation, "total_leaves_encashed")
-			+ self.encashment_days,
-		)
+		if self.pay_via_payment_entry:
+			self.create_gl_entries()
+		else:
+			additional_salary = frappe.new_doc("Additional Salary")
+			additional_salary.company = frappe.get_value("Employee", self.employee, "company")
+			additional_salary.employee = self.employee
+			additional_salary.currency = self.currency
+			earning_component = frappe.get_value("Leave Type", self.leave_type, "earning_component")
+			if not earning_component:
+				frappe.throw(_("Please set Earning Component for Leave type: {0}.").format(self.leave_type))
+			additional_salary.salary_component = earning_component
+			additional_salary.payroll_date = self.encashment_date
+			additional_salary.amount = self.encashment_amount
+			additional_salary.ref_doctype = self.doctype
+			additional_salary.ref_docname = self.name
+			additional_salary.submit()
+
+			# Set encashed leaves in Allocation
+			frappe.db.set_value(
+				"Leave Allocation",
+				self.leave_allocation,
+				"total_leaves_encashed",
+				frappe.db.get_value("Leave Allocation", self.leave_allocation, "total_leaves_encashed")
+				+ self.encashment_days,
+			)
```

Comment:
- This is applicable for both pay via payment entry and salary?
- create a separate function `set_encashed_leaves_in_leave_allocation`